src/windy/platforms/win32/platform.nim

```diff
@@ -35,15 +35,22 @@ type
     onResize*: Callback
     onFocusChange*: Callback
     onMouseMove*: Callback
-    onScroll*: ScrollCallback
+    onScroll*: Callback
+    onButtonPress*: ButtonCallback
+    onButtonRelease*: ButtonCallback
 
-    mouse: Mouse
+    perFrame: PerFrame
     trackMouseEventRegistered: bool
+    mousePos: IVec2
+    buttonPressed, buttonDown, buttonReleased: BitArray
 
     hWnd: HWND
     hdc: HDC
     hglrc: HGLRC
 
+  ButtonView* = object
+    states: BitArray
```

Comment:
Using a set would be perfectly reasonable.

A nice thing about a bit array is this: since we know exactly how many entries there are (there will end up being hundreds of button codes, something like that), we can set up the memory once and never need to do anything but read and write to it after. With a set, it needs to hash and maintain hash buckets that may need to be resized. Not a big deal at all in this case, but not necessary either.

We will see what the "stable" API settles on. It may be sets, since there is an argument for enabling Windy users to override / write the button states if they want to in their program. Using the standard library set will make that more familiar vs a bit array.

Reply:
https://github.com/treeform/bitty/pull/13 includes a why. Bitty appears 4x faster and uses 25x less memory. I do not present that as a perfect comparison but it presents hard data about what I mean.

This isn't really a hot path so performance is not the only consideration. However, as things stand, why use sets when they are both slower and more bloated, if there is no benefit to it.

Reply:
Ah, I bet you mean Nim's `set`, not the package `sets`. Strange of Nim to have this all over the place. In that case, using Nim's `set` should be nearly identical to Bitty and there probably is no reason not to use the standard lib version.